packages/shared/src/components/ShareOptionsMenu.tsx

```diff
@@ -0,0 +1,215 @@
+      className: 'bg-theme-color-bun',
+      label: `${post?.bookmarked ? 'Remove from' : 'Save to'} bookmarks`,
+      action: onBookmark,
+    },
+    {
+      icon: LinkIcon,
+      className: 'bg-theme-color-pepper',
+      label: 'Copy link to article',
+      action: trackAndCopyLink,
+    },
+    {
+      href: getTwitterShareLink(link, post?.title),
+      icon: TwitterIcon,
+      className: 'bg-theme-bg-twitter',
+      onClick: () => trackClick(ShareProvider.Twitter),
+      label: 'Twitter',
+    },
+    {
+      href: getWhatsappShareLink(link),
+      icon: WhatsappIcon,
+      onClick: () => trackClick(ShareProvider.WhatsApp),
+      className: 'bg-theme-bg-whatsapp',
+      label: 'WhatsApp',
+    },
+    {
+      href: getFacebookShareLink(link),
+      icon: FacebookIcon,
+      className: 'bg-theme-bg-facebook',
+      onClick: () => trackClick(ShareProvider.Facebook),
+      label: 'Facebook',
+    },
+    {
+      href: getRedditShareLink(link, post?.title),
+      icon: RedditIcon,
+      className: 'bg-theme-bg-reddit',
+      onClick: () => trackClick(ShareProvider.Reddit),
+      label: 'Reddit',
+    },
+    {
+      href: getLinkedInShareLink(link),
+      icon: LinkedInIcon,
+      className: 'bg-theme-bg-linkedin',
+      onClick: () => trackClick(ShareProvider.LinkedIn),
+      label: 'LinkedIn',
+    },
+    {
+      href: getTelegramShareLink(link, post?.title),
+      icon: TelegramIcon,
+      className: 'bg-theme-bg-telegram',
+      onClick: () => trackClick(ShareProvider.Telegram),
+      label: 'Telegram',
+    },
+  ];
+  if (postCardShareVersion === ShareVersion.V2) {
+    ShareOptions.push(
+      ...[
+        {
+          icon: <MenuIcon Icon={ShareIcon} />,
+          text: 'Share article via...',
+          action: onShare,
+        },
+        {
+          icon: <MenuIcon secondary={post?.bookmarked} Icon={BookmarkIcon} />,
+          text: `${post?.bookmarked ? 'Remove from' : 'Save to'} bookmarks`,
+          action: onBookmark,
+        },
+        {
+          icon: <MenuIcon Icon={LinkIcon} />,
+          text: 'Copy link to article',
+          action: trackAndCopyLink,
+        },
+      ],
+    );
+  }
+  if (postCardShareVersion === ShareVersion.V3) {
+    ShareOptions.push(
+      ...[
+        ...socials.map((social) => ({
+          href: social.href,
+          icon: (
+            <MenuIconWithBg
+              secondary={social.secondary}
+              className={social.className}
+              Icon={social.icon}
+            />
+          ),
+          text: social.label,
+          action: social.onClick,
+        })),
+      ],
+    );
+  }
+  if (postCardShareVersion === ShareVersion.V4) {
+    ShareOptions.push(
+      ...[
+        ...socials.map((social) => ({
+          href: social.href,
+          icon: <MenuIcon secondary={social.secondary} Icon={social.icon} />,
+          text: social.label,
+          action: social.onClick,
+        })),
+      ],
```

Comment:
The brackets shouldn't be necessary.
```suggestion
      ...socials.map((social) => ({
        href: social.href,
        icon: <MenuIcon secondary={social.secondary} Icon={social.icon} />,
        text: social.label,
        action: social.onClick,
      })),
```